### PR TITLE
feat(zsh): zsh pgo

### PR DESCRIPTION
**Pull request by lseman**
zsh pgo, but we will wait until finally merging it;
I'm testing it on daily bassis as my main shell, we need to make sure nothing breaks on the user shell and it is 100% working.

### CODE REVIEW

**Comment by lseman** [comment]
Is is working well, not a single problem.

**Review by ventureoo — APPROVED** [review]
LGTM

File: zsh/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    # force generation of documentation with correct paths
+    rm Doc/version.yo
+}
+
+build() {
+    cd "${srcdir}/${pkgbase}-${pkgver}"
+
+    export CC=gcc
+    export CXX=g++
+    export LIBS="-lgcov"
```
Comment (vnepogodin) [inline zsh/PKGBUILD:59]:
```suggestion
```

File: zsh/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        --with-tcsetpgrp \
+        --enable-pcre \
+        --enable-gdbm \
+        --enable-cap \
+        --enable-zsh-secure-free
+
+    make
+    cd Test
+    ZTST_continue=1 make test || true
+    cd ..
```
Comment (vnepogodin) [inline zsh/PKGBUILD:88]:
```suggestion
    HOME="${srcdir}" make check || true
```

File: zsh/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+build() {
+    cd "${srcdir}/${pkgbase}-${pkgver}"
+
+    export CC=gcc
+    export CXX=g++
+    export LIBS="-lgcov"
+
+    export CFLAGS="${CFLAGS} -fprofile-generate -fprofile-update=atomic -fprofile-partial-training"
+    export CXXFLAGS="${CXXFLAGS} -fprofile-generate -fprofile-update=atomic -fprofile-partial-training"
```
Comment (vnepogodin) [inline zsh/PKGBUILD:62]:
```suggestion
    export CFLAGS="${CFLAGS} -fprofile-dir=${PROFILES_PGO_DIR} -fprofile-generate=${PROFILES_PGO_DIR} -fprofile-update=atomic -fprofile-partial-training"
    export CXXFLAGS="${CXXFLAGS} -fprofile-dir=${PROFILES_PGO_DIR} -fprofile-generate=${PROFILES_PGO_DIR} -fprofile-update=atomic -fprofile-partial-training"
```

File: zsh/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    ZTST_continue=1 make test || true
+    cd ..
+
+    export CFLAGS="${CFLAGS/-fprofile-generate/}"
+    export CXXFLAGS="${CXXFLAGS/-fprofile-generate/}"
+    export CFLAGS="${CFLAGS//-fprofile-update=atomic/}"
+    export CXXFLAGS="${CXXFLAGS//-fprofile-update=atomic/}"
+
+    export CFLAGS="${CFLAGS} -fprofile-use -Wno-error=coverage-mismatch"
+    export CXXFLAGS="${CXXFLAGS} -fprofile-use -Wno-error=coverage-mismatch"
```
Comment (vnepogodin) [inline zsh/PKGBUILD:96]:
```suggestion
    export CFLAGS="${CFLAGS} -fprofile-dir=${PROFILES_PGO_DIR} -fprofile-use=${PROFILES_PGO_DIR} -Wno-error=coverage-mismatch"
    export CXXFLAGS="${CXXFLAGS} -fprofile-dir=${PROFILES_PGO_DIR} -fprofile-use=${PROFILES_PGO_DIR} -Wno-error=coverage-mismatch"
```

File: zsh/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    unset LIBS
```
Comment (vnepogodin) [inline zsh/PKGBUILD:97]:
```suggestion
```

File: zsh/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    done
+    rm Completion/Linux/Command/_pkgtool
+
+    # force generation of documentation with correct paths
+    rm Doc/version.yo
+}
+
+build() {
+    cd "${srcdir}/${pkgbase}-${pkgver}"
+
```
Comment (vnepogodin) [inline zsh/PKGBUILD:56]:
```suggestion
    export PROFILES_PGO_DIR="$PWD/pgo-profiles"
    mkdir -p "${PROFILES_PGO_DIR}"
```

**Review by vnepogodin — CHANGES_REQUESTED** [review]

**Pull request merged**